api/envoy/config/route/v3/route_components.proto

```diff
@@ -436,6 +436,10 @@ message WeightedCluster {
   // configuration file will be used as the default weight. See the :ref:`runtime documentation
   // <operations_runtime>` for how key names map to the underlying implementation.
   string runtime_key_prefix = 2;
+
+  // Specifies the key to look up the random value that is used in weighted cluster picking algorithm.
+  // This enables providing the random value through configuration.
+  string random_value_key = 4;
```

Comment:
On an API level I would think that we'd want this to be a bit more extensible to support future use cases, so I'd put this into a `oneof` at the very least. @envoyproxy/api-shepherds can advice here 

"Key" is ambiguous here, I think we should explicit call this a header value or use the HeaderMatcher proto to decide which proto to use

Reply:
+1 I think we also need to clarify the comment that this is header name that will have a number for weight selection. Also include what happens if the header is present but is not a number.

Reply:
Yeah, this value could come from other sourcese, e.g. dynamic metadata or filter state, in the future.

Reply:
`oneof` is a good idea! Future thinking, besides `header_name` here, a simple use case is that we can provided random value directly via config (i.e., int64 random_value).

PTAL. Thanks!

PS: The doc change to current.est will be added at last before submitting once the PR is LGTMed